src/cpp/handler/handlerengine.cpp

```diff
@@ -1248,6 +1252,12 @@ class HandlerEngine::Private : public QObject
 	Connection controlReqReadyConnection;
 	Connection controlServerConnection;
 	Connection itemReadyConnection;
+	map<Deferred*, Connection> sessionCreateOrUpdateFinishedConnection;
+	map<Deferred*, Connection> finishedConnection;
+	map<Deferred*, Connection> deferredFinishedConnection;
+	map<Deferred*, Connection> deferredFinishedConn;
+	map<Deferred*, Connection> wFinishedConnection;
+	map<Deferred*, Connection> awFinishedConnection;
```

Comment:
Maybe everything could go into one map, `finishedConnection`?

Reply:
great point! I merged `wFinishedConnection` and `aWFinishedConnection` but the rest can not be consolidated as they could have the same deferred as map keys unless we make the map key more complex which might result in issues when it's time to erase.

Reply:
Duplicate `Deferred` pointers for different objects should not be possible even if the actual derived types vary, as they always point to unique objects. So I think as long as we don't need to search the map by type (e.g. "find me all SessionCreateOrUpdate deferreds") then one map ought to do.

Reply:
got it thanks. these maps are consolidated.